test/framework/clusterresourceset_helpers.go

```diff
@@ -129,23 +131,21 @@ func WaitForClusterResourceSetToApplyResources(ctx context.Context, input WaitFo
 		Expect(input.ClusterProxy.GetClient().Get(ctx, types.NamespacedName{Name: input.Cluster.Name, Namespace: input.Cluster.Namespace}, binding)).To(Succeed())
 
 		for _, resource := range input.ClusterResourceSet.Spec.Resources {
-			if resource.Kind == string(addonsv1.SecretClusterResourceSetResourceKind) {
-				secret := &corev1.Secret{}
-				err := input.ClusterProxy.GetClient().Get(ctx, types.NamespacedName{Name: resource.Name, Namespace: input.ClusterResourceSet.Namespace}, secret)
-				if err == nil {
-
-					if !binding.Spec.Bindings[0].IsApplied(resource) {
-						return false
-					}
-				}
-			} else if resource.Kind == string(addonsv1.ConfigMapClusterResourceSetResourceKind) {
-				configMap := &corev1.ConfigMap{}
-				err := input.ClusterProxy.GetClient().Get(ctx, types.NamespacedName{Name: resource.Name, Namespace: input.ClusterResourceSet.Namespace}, configMap)
-				if err == nil {
-					if !binding.Spec.Bindings[0].IsApplied(resource) {
-						return false
-					}
-				}
+			var configSource runtime.Object
+
+			switch resource.Kind {
+			case string(addonsv1.SecretClusterResourceSetResourceKind):
+				configSource = &corev1.Secret{}
+			case string(addonsv1.ConfigMapClusterResourceSetResourceKind):
+				configSource = &corev1.ConfigMap{}
+			}
+
+			if err := input.ClusterProxy.GetClient().Get(ctx, types.NamespacedName{Name: resource.Name, Namespace: input.ClusterResourceSet.Namespace}, configSource); err != nil {
+				continue
```

Comment:
```suggestion
				// If the resource is missing, CRS will not requeue but retry at each reconcile, 
				// because this is not an error. So, we are only interested in seeing the resources that exist to be applied by CRS.
				continue
```